src/infra/auth/jwt.ts

```diff
@@ -85,15 +88,28 @@ let $getUserTokenDataCallback: GetUserTokenDataFn = async (
 
 export const createSessionToken = async (
 	userId: number,
-	{ existingToken, jwtOptions }: ExtraParams = {},
+	{ existingToken, jwtOptions, tx }: ExtraParams = {},
 ): Promise<string> => {
-	const tokenData = await $getUserTokenDataCallback(userId, existingToken);
+	const tokenData = await $getUserTokenDataCallback(userId, existingToken, tx);
 	return createJwt(tokenData, jwtOptions);
 };
 
-const sendXHRToken = (res: Response, token: string, statusCode = 200): void => {
-	res.header('content-type', 'text/plain');
-	res.status(statusCode).send(token);
+const sendXHRToken = (
+	res: Response,
+	token: string,
+	statusCode = 200,
+	tx?: Tx,
+) => {
+	const $sendXHRToken = () => {
+		res.header('content-type', 'text/plain');
+		res.status(statusCode).send(token);
+	};
+
+	if (tx) {
+		tx.on('end', $sendXHRToken);
```

Comment:
Otherwise we would be sending the response before the transaction has fully committed.
This is significant enough to have fasttest of the cloud API fail w/o it.

Reply:
A comment explaining the need should be in the code for future reference (rather than needing to look back at the PR comments)